src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/BookingSearchServiceTest.kt

```diff
@@ -320,28 +514,289 @@ class BookingSearchServiceTest {
           .produce()
       }
       .produce()
+    every { mockBookingRepository.findBookings(any(), any(), any(), any()) } returns PageImpl(
+      listOf(
+        TestBookingSearchResult(),
+        TestBookingSearchResult(),
+        TestBookingSearchResult(),
+      ),
+    )
+    every { mockOffenderService.getOffenderByCrn(any(), any()) } returns AuthorisableActionResult.Success(
+      OffenderDetailsSummaryFactory().produce(),
+    )
+
+    val (result, metaData) = bookingSearchService.findBookings(
+      ServiceName.temporaryAccommodation,
+      null,
+      sortOrder,
+      BookingSearchSortField.bookingCreatedAt,
+      null,
+    )
+
+    assertThat(result is AuthorisableActionResult.Success).isTrue
+    assertThat(result.entity is ValidatableActionResult.Success).isTrue
+    val validationResult = result.entity as ValidatableActionResult.Success
+    assertThat(validationResult.entity).hasSize(3)
+    assertThat(validationResult.entity.map { it.bookingCreatedAt }).isSortedAccordingTo { a, b ->
+      when (sortOrder) {
+        SortOrder.ascending -> compareValues(a, b)
+        SortOrder.descending -> compareValues(b, a)
+      }
+    }
+    assertThat(metaData).isNull()
+    verify(exactly = 1) {
+      mockBookingRepository.findBookings(any(), any(), any(), any())
+    }
+    verify(exactly = 3) {
+      mockOffenderService.getOffenderByCrn(any(), any())
+    }
+  }
+
+  @EnumSource(value = SortOrder::class)
+  @ParameterizedTest
+  fun `findBookings returns results not sorted according to booking creation when page number is given`(sortOrder: SortOrder) {
```

Comment:
I'm not sure I understand why we want to prevent sorting by the booking creation date when using pagination.

Reply:
When the page is not given, then we dont use the query sort. We only retrieve data and sort it in the code (which was existing function). This is to support API backward compatible until UI change.

Reply:
When we don't get any query parameters for pagination, we could setup a `Pageable` to return a single page using a page size of `Integer.MAX_VALUE`. That way we'd be able to use the sorting functionality of the database instead of doing it in memory.

Reply:
done